src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/GeneralLetterController.java

```diff
@@ -16,19 +16,73 @@
 import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
 import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
 import uk.gov.hmcts.reform.finrem.caseorchestration.service.GeneralLetterService;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.IdamService;
 
 import javax.validation.constraints.NotNull;
 
+import java.util.Map;
+
 import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
 import static uk.gov.hmcts.reform.finrem.caseorchestration.OrchestrationConstants.AUTHORIZATION_HEADER;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.GENERAL_LETTER_ADDRESS_TO;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.GENERAL_LETTER_BODY;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.GENERAL_LETTER_PREVIEW;
 
 @RestController
 @RequestMapping(value = "/case-orchestration")
 @RequiredArgsConstructor
 @Slf4j
 public class GeneralLetterController implements BaseController {
 
-    private final GeneralLetterService service;
+    private final IdamService idamService;
+    private final GeneralLetterService generalLetterService;
+
+    @PostMapping(path = "/general-letter-start", consumes = APPLICATION_JSON_VALUE, produces = APPLICATION_JSON_VALUE)
```

Comment:
it looks like you've removed this from you CCD PR, I would say this should be done as part of the submit rather than the start journey in case the user decided to cancel

Reply:
this was back-end refactoring, CCD config should stay as it is regarding that part - it is clearing fields carrying data across screens for general letter. I.e. there are fields like general letter address to which are not really part of case data, but are still carried around so that when the event is submitted general letter document can be generated.